### PR TITLE
Shorten table names and re-enable app code checker

### LINKED ISSUE #784: Shorten table names and re-enable app code checker [$80]
Now that Nextcloud support database migration, it should be easy to shorten our table names to be app code checker compliant again.

To do:
- [ ] DB migration class
- [ ] Tests -  if migrations are testable at all
- [ ] Bump version number
- [ ] Update docs that point to old table names
- [ ] Re-enable app code checker on CI

<bountysource-plugin>

---
There is a **[$80 open bounty](https://www.bountysource.com/issues/55081952-shorten-table-names-and-re-enable-app-code-checker?utm_campaign=plugin&utm_content=tracker%2F44154351&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F44154351&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by danielkesselberg**
Close #784

### CODE REVIEW

**Comment by ChristophWurst** [comment]
Wow, impressive work! Will try to integrate this ASAP. Took a quick look at the code and it looks very good :rocket:

**Comment by danielkesselberg** [comment]
Cool. Let me know if there is anything else to do.

**Comment by ChristophWurst** [comment]
Updated with latest changes on master.

Gave this a local test. Migration ran smooth, no data was lost as far as I can tell and CI was happy with the last run. Will merge when all goes green again :rocket:

**Comment by danielkesselberg** [comment]
You mentioned in #784 that a migration with many data might run out of memory. Is there anything we could do to avoid this?  I guess we could move copy addresses logic into a own command and call this from migration when count(addresses) < 100.000 and throw a warning otherwise.

**Comment by ChristophWurst** [comment]
> You mentioned in #784 that a migration with many data might run out of memory. Is there anything we could do to avoid this? I guess we could move copy addresses logic into a own command and call this from migration when count(addresses) < 100.000 and throw a warning otherwise.

Something like https://github.com/nextcloud/mail/blob/d86a200b15bdad41c3c0ced26b85797b44184cd5/lib/Migration/FixCollectedAddresses.php#L47-L62.

**Comment by ChristophWurst** [comment]
Oh, since this targets the very same table, we might be able to re-use that chunking logic: https://github.com/nextcloud/mail/blob/d86a200b15bdad41c3c0ced26b85797b44184cd5/lib/Db/CollectedAddressMapper.php#L83-L99. Of course, this only works if the mapper accesses the old table …

**Comment by danielkesselberg** [comment]
> Of course, this only works if the mapper accesses the old table …

image
:see_no_evil:

**Comment by ChristophWurst** [comment]
> see_no_evil

Actually, since Nextcloud 14 is php7+, this is a valid approach. Go for it :wink:

**Comment by danielkesselberg** [comment]
More complicated than I thought :confused: 

I found something while testing the migration. https://github.com/nextcloud/mail/blob/master/lib/Migration/FixCollectedAddresses.php#L66
<pre>/home/vagrant/code/nextcloud/apps/mail/lib/Migration/FixCollectedAddresses.php:66:
string(4) &quot;NULL&quot;
/home/vagrant/code/nextcloud/apps/mail/lib/Migration/FixCollectedAddresses.php:66:
string(4) &quot;NULL&quot;
/home/vagrant/code/nextcloud/apps/mail/lib/Migration/FixCollectedAddresses.php:66:
string(4) &quot;NULL&quot;
</pre>

I guess the condition is never true because "NULL" is a string. I wond

**Comment by ChristophWurst** [comment]
> I guess the condition is never true because "NULL" is a string

Ouch. Is it possible this happen during the development of the migration? As in copying some data wrongly made real `null` values `NULL` strings?

**Comment by danielkesselberg** [comment]
> > I guess the condition is never true because "NULL" is a string
> 
> Ouch. Is it possible this happen during the development of the migration? As in copying some data wrongly made real `null` values `NULL` strings?

I inserted "null" into the table :laughing: :rofl: With a real null everything works as expected :+1:

File: lib/Migration/Version0100Date20180825194217.php [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+ */
+
+namespace OCA\Mail\Migration;
+
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\SimpleMigrationStep;
+use OCP\Migration\IOutput;
+
+/**
+ * Auto-generated migration step: Please modify to your needs!
```
Comment (ChristophWurst) [inline lib/Migration/Version0100Date20180825194217.php]: please remove this comment

File: lib/Migration/Version0100Date20180825194217.php [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+ */
+class Version0100Date20180825194217 extends SimpleMigrationStep {
+
+	/**
+	 * @param IOutput $output
+	 * @param \Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function preSchemaChange(IOutput $output, \Closure $schemaClosure, array $options) {
```
Comment (ChristophWurst) [inline lib/Migration/Version0100Date20180825194217.php]: remove empty override

File: lib/Migration/Version0100Date20180825194217.php [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+		return $schema;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param \Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function postSchemaChange(IOutput $output, \Closure $schemaClosure, array $options) {
```
Comment (ChristophWurst) [inline lib/Migration/Version0100Date20180825194217.php]: remove empty override

File: lib/Migration/Version0110Date20180825195812.php [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+ * Auto-generated migration step: Please modify to your needs!
+ */
+class Version0110Date20180825195812 extends SimpleMigrationStep {
+
+	/**
+	 * @param IOutput $output
+	 * @param \Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function preSchemaChange(IOutput $output, \Closure $schemaClosure, array $options) {
```
Comment (ChristophWurst) [inline lib/Migration/Version0110Date20180825195812.php]: remove empty override

File: lib/Migration/Version0110Date20180825195812.php [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+		return $schema;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param \Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function postSchemaChange(IOutput $output, \Closure $schemaClosure, array $options) {
```
Comment (ChristophWurst) [inline lib/Migration/Version0110Date20180825195812.php]: remove empty override

File: lib/Migration/Version0110Date20180825201241.php [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+		return $schema;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param \Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function postSchemaChange(IOutput $output, \Closure $schemaClosure, array $options) {
```
Comment (ChristophWurst) [inline lib/Migration/Version0110Date20180825201241.php]: remove empty override

**Comment by ChristophWurst** [comment]
> [ @danielkesselberg ](/danielkesselberg)
> ` [art](/nextcloud/mail/pull/1061/commits/0be69be250203597f7bd7d40751f53e6eb1304c2) `

Yes, coding is :art:.

**Comment by ChristophWurst** [comment]
Hey @danielkesselberg,

just found out there's a pending bounty on the ticket this PR fixed. Please see https://www.bountysource.com/issues/55081952-shorten-table-names-and-re-enable-app-code-checker and claim it. Thanks :v:

**Comment by ChristophWurst** [comment]
@danielkesselberg could you please confirm https://www.bountysource.com/people/62764-daniel512 is you, then I'll accept the claim.

**Comment by danielkesselberg** [comment]
Its me :smile:

**Pull request merged**